File: java_solutions/2444.count-subarrays-with-fixed-bounds.java
/*
 * @lc app=leetcode id=2444 lang=java
 *
 * [2444] Count Subarrays With Fixed Bounds
 */

class Solution { public long countSubarrays(int[] nums, int minK, int maxK) { return ((Function<int[], Long>) (state -> IntStream.range(0, nums.length).mapToLong(i -> (nums[i] < minK || nums[i] > maxK ? (state[0] = i) : 0) * 0L + (nums[i] == minK ? (state[1] = i) : 0) * 0L + (nums[i] == maxK ? (state[2] = i) : 0) * 0L + Math.max(0, Math.min(state[1], state[2]) - state[0])).sum())).apply(new int[]{-1, -1, -1}); } }

File: kotlin_solutions_1/2444.count-subarrays-with-fixed-bounds.kt
/*
 * @lc app=leetcode id=2444 lang=kotlin
 *
 * [2444] Count Subarrays With Fixed Bounds
 */
class Solution { fun countSubarrays(nums: IntArray, minK: Int, maxK: Int): Long = nums.indices.fold(longArrayOf(0, -1, -1, -1)) { a, i -> a.apply { if (nums[i] < minK || nums[i] > maxK) set(3, i.toLong()) }.apply { if (nums[i] == minK) set(1, i.toLong()) }.apply { if (nums[i] == maxK) set(2, i.toLong()) }.apply { set(0, get(0) + kotlin.math.max(0L, kotlin.math.min(get(1), get(2)) - get(3))) } }.let { it[0] } }